### PR TITLE
Document all important env vars

### PR DESCRIPTION
**Pull request by peterbe**
Part of #575

It's a start. 
It doesn't close the issue because there are probably things I've forgotten. 
Also, the `BUILD_FLAWS` one is deliberately omitted because it's due to change pretty soon.

### CODE REVIEW

**Comment by peterbe** [comment]
@escattone ping?

File: docs/envvars.md [hunk 1]
```diff
@@ -5,6 +5,17 @@ specific to the client and some are specific to the builder etc.
...
+## General
+
+### `EDITOR`
+
+#### Default: n/a
+
+This is a general OS, optional, OS environment variable and it's needed
+if you want the ability to go from viewing a rendered document to
+open the source in your preferred editor. It's needed for the
+"Edit this page in your editor" button to work.
```
Comment (escattone) [inline docs/envvars.md:17]:
```suggestion
This is a general OS, optional, environment variable and it's needed
if you want the ability to go from viewing a rendered document to
opening the source in your preferred editor. It's needed for the
"Edit this page in your editor" button to work.
```

File: docs/envvars.md [hunk 2]
```diff
@@ -13,6 +24,25 @@ CLI arguments. And
...
+
+**Default: `client/build`**
+
+Where the built `index.html` and `index.json` files go.
+
+### `BUILD_POPULARITIES_FILEPATH`
+
+**Default: `content/popularities.json`**
+
+It's the location to a .json file that has a popularities based on
```
Comment (escattone) [inline docs/envvars.md:43]:
```suggestion
It's the location to a .json file that has popularities based on
```

File: docs/envvars.md [hunk 3]
```diff
@@ -27,3 +57,47 @@ Generating this can be time-consuming if you're doing rapid development.
...
+
+**Default: `[]`** (meaning, all)
+
+**Example: `en-us,fr`**
+
+**Note** It's a comma separated string.
+
+Limits which locales to build when not specifying a specific folder or a
+specific locale. Can be useful when you often build large batches but
+is aware that you only care about these specific locales.
```
Comment (escattone) [inline docs/envvars.md:71]:
```suggestion
only care about these specific locales.
```

**Pull request merged**